chain/chain/src/store.rs

```diff
@@ -2333,124 +2333,142 @@ impl<'a> ChainStoreUpdate<'a> {
         Ok(chain_store_update)
     }
 
+    #[tracing::instrument(
+        level = "debug",
+        target = "store",
+        "ChainUpdate::finalize",
+        skip_all,
+    )]
     fn finalize(&mut self) -> Result<StoreUpdate, Error> {
         let mut store_update = self.store().store_update();
-        Self::write_col_misc(&mut store_update, HEAD_KEY, &mut self.head)?;
-        Self::write_col_misc(&mut store_update, TAIL_KEY, &mut self.tail)?;
-        Self::write_col_misc(&mut store_update, CHUNK_TAIL_KEY, &mut self.chunk_tail)?;
-        Self::write_col_misc(&mut store_update, FORK_TAIL_KEY, &mut self.fork_tail)?;
-        Self::write_col_misc(&mut store_update, HEADER_HEAD_KEY, &mut self.header_head)?;
-        Self::write_col_misc(&mut store_update, FINAL_HEAD_KEY, &mut self.final_head)?;
-        Self::write_col_misc(
-            &mut store_update,
-            LARGEST_TARGET_HEIGHT_KEY,
-            &mut self.largest_target_height,
-        )?;
-        debug_assert!(self.chain_store_cache_update.blocks.len() <= 1);
-        for (hash, block) in self.chain_store_cache_update.blocks.iter() {
-            let mut map = HashMap::clone(
-                self.chain_store.get_all_block_hashes_by_height(block.header().height())?.as_ref(),
-            );
-            map.entry(block.header().epoch_id().clone())
-                .or_insert_with(|| HashSet::new())
-                .insert(*hash);
-            store_update.set_ser(
-                DBCol::BlockPerHeight,
-                &index_to_bytes(block.header().height()),
-                &map,
+        {
```

Comment:
@nagisa please let me know if I should revert the tracings inside this function. Definitely helps a lot when I investigate what exactly is slow inside

Reply:
They are fine, but as in the other comment -- lets downgrade the level to `trace` on everything in this file.